e2e_test/batch/basic/scientific_number.slt

```diff
@@ -0,0 +1,21 @@
+statement ok
+SET RW_IMPLICIT_FLUSH TO true;
+
+query IIRR
+SELECT 1e6, -1e6, 1e-6, -1e-6, -1.25e3, 1.25e3
```

Comment:
Thanks for your contribution!
We can add some test cases where the number overflows or underflows and thus returns an error,
but we can also do it later